### PR TITLE
Add Schema.subtypes to t-wasm

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

File: trustfall_wasm/src/trustfall_wasm.d.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export interface Adapter<T> {
 
 export class Schema {
     free(): void;
+    subtypes(type_name: string): Set<string>
```
Comment (obi1kenobi) [inline trustfall_wasm/src/trustfall_wasm.d.ts:54]: Could you tweak this to add the type signature information and match the surrounding style here?

File: trustfall_wasm/src/lib.rs [hunk 2]
```diff
@@ -50,6 +51,21 @@ impl Schema {
             .map(Schema::new)
             .map_err(crate::InvalidSchemaError::new)
     }
+
+    pub fn subtypes(&self, type_name: &str) -> Set {
+        let subtypes_iter = match self.0.subtypes(type_name) {
+            Some(iter) => iter,
+            None => panic!("type {type_name} is not part of this schema"),
+        };
```
Comment (obi1kenobi) [inline trustfall_wasm/src/lib.rs:59]: This might be a bit more idiomatic in Rust with `.unwrap_or_else(|| panic!(...))`

**Review by obi1kenobi — COMMENTED** [review]
Two minor nitpicks, otherwise this looks great and I'm happy to merge it!

File: trustfall_wasm/src/lib.rs (hunk 2 above)
Comment (u9g) [reply trustfall_wasm/src/lib.rs]: done in dd7ffda

File: trustfall_wasm/src/trustfall_wasm.d.ts (hunk 1 above)
Comment (u9g) [reply trustfall_wasm/src/trustfall_wasm.d.ts:64]: done in 85d5a64

File: trustfall_wasm/src/trustfall_wasm.d.ts [hunk 3]
```diff
@@ -51,6 +51,17 @@ export interface Adapter<T> {
 
 export class Schema {
     free(): void;
+    /**
```
Comment (obi1kenobi) [inline trustfall_wasm/src/trustfall_wasm.d.ts:54]:
```suggestion

    /**
```

File: trustfall_wasm/src/trustfall_wasm.d.ts [hunk 4]
```diff
@@ -51,6 +51,17 @@ export interface Adapter<T> {
...
+     * Returns a Set<string> which contains every subtype of type_name
+     * in the schema and itself. This allows you to get the subtypes using
+     * this function, then in resolveCoercion return whether the set returned
+     * by this function has the typename string of the vertex from the iterator,
+     * which would mean coercible from it's current form to the form wanted by
+     * resolveCoercion.
```
Comment (obi1kenobi) [inline trustfall_wasm/src/trustfall_wasm.d.ts:60]: Good function descriptions are hard, let's try to keep them concise while still communicating necessary information. Here's a suggestion:
```suggestion
     * Returns a `Set<string>` naming all subtypes (plus itself) of `type_name` in the schema.
     *
     * May be used in `Adapter.resolveCoercion` to determine if vertices can be coerced
     * to the specified type.
```

File: trustfall_wasm/src/trustfall_wasm.d.ts [hunk 5]
```diff
@@ -51,6 +51,17 @@ export interface Adapter<T> {
...
+     * @param {string} type_name
+     * @returns {Set<string>}
```
Comment (obi1kenobi) [inline trustfall_wasm/src/trustfall_wasm.d.ts:62]:
```suggestion
     * @returns {Set<string>}
     * @throws if `type_name` is not an interface or object type in this schema.
```

File: trustfall_wasm/src/trustfall_wasm.d.ts (hunk 1 above)
Comment (obi1kenobi) [reply trustfall_wasm/src/trustfall_wasm.d.ts:64]:
```suggestion
    subtypes(type_name: string): Set<string>;
```

**Review by obi1kenobi — COMMENTED** [review]
A few nitpicks on things that slipped through the cracks.

**Comment by obi1kenobi** [comment]
If you're fine with the suggestions, I'm happy to merge this.

**Comment by u9g** [comment]
> If you're fine with the suggestions, I'm happy to merge this.

Happily commited :)

File: trustfall_wasm/src/trustfall_wasm.d.ts (hunk 4 above)
Comment (u9g) [reply trustfall_wasm/src/trustfall_wasm.d.ts]: This is great, thanks!

**Review by obi1kenobi — APPROVED** [review]

**Pull request merged**